test/LivepeerProtocol.js

```diff
@@ -1192,4 +1197,501 @@ contract('LivepeerProtocol', function(accounts) {
+
+            // Account 1 bonds to self as transcoder
+            await instance.bond(a1Stake, accounts[1], {from: accounts[1]});
+
+            // Fast forward 1 round
+            await rpc.waitUntilNextBlockMultiple(20, ROUND_LENGTH);
+
+            await instance.initializeRound();
+
+            // Account 2 creates a job
+            await instance.job(1, "0x1", 200, {from: accounts[2]});
+
+            // Account 1 claims work
+            await instance.claimWork(0, 0, 3, "0x1", {from: accounts[1]});
+
+            // Account 2 ends the job
+            await instance.endJob(0, {from: accounts[2]});
+
+            // Fast forward through job ending period
+            await rpc.wait(20, JOB_ENDING_PERIOD);
+
+            let threw = false;
+
+            try {
+                // Account 1 calls verify
+                // Fake param data
+                await instance.verify(0, 2, "0x1", "0x2", "0x3", "0x4", {from: accounts[1]});
+            } catch (err) {
+                threw = true;
+            }
+
+            assert.isOk(threw, "verify did not throw for an inactive job");
+        });
+
+        it("should fail if the sender is not the assigned transcoder for the job", async function() {
+            const instance = await LivepeerProtocol.new(2, ROUND_LENGTH, CYCLES_PER_ROUND, {from: accounts[0]});
+            const lptaddress = await instance.token.call();
+            const lpt = await LivepeerToken.at(lptaddress);
+
+            const a1Stake = 2000;
+
+            // Transfer tokens
+            await lpt.transfer(accounts[1], toSmallestUnits(1), {from: accounts[0]});
+
+            // Register account 1 as transcoder 1
+            await instance.transcoder(BLOCK_REWARD_CUT, FEE_SHARE, PRICE_PER_SEGMENT, {from: accounts[1]});
+
+            // Approve token transfer for account 1
+            await lpt.approve(instance.address, a1Stake, {from: accounts[1]});
+
+            // Account 1 bonds to self as transcoder
+            await instance.bond(a1Stake, accounts[1], {from: accounts[1]});
+
+            // Fast forward 1 round
+            await rpc.waitUntilNextBlockMultiple(20, ROUND_LENGTH);
+
+            await instance.initializeRound();
+
+            // Account 2 creates a job
+            await instance.job(1, "0x1", 200, {from: accounts[2]});
+
+            // Account 1 claims work
+            await instance.claimWork(0, 0, 3, "0x1", {from: accounts[1]});
+
+            let threw = false;
+
+            try {
+                // Account 3 calls verify
+                // Fake param data
+                await instance.verify(0, 2, "0x1", "0x2", "0x3", "0x4", {from: accounts[3]});
+            } catch (err) {
+                threw = true;
+            }
+
+            assert.isOk(threw, "verify did not throw when sender is not the assigned transcoder for the job");
+        });
+
+        it("should fail if the segment is not eligible for verification", async function() {
+
```

Comment:
This is left blank for now because still figuring out the best way to test segment verification eligibility